src/dfx/src/commands/generate.rs

```diff
@@ -31,40 +33,62 @@ pub fn exec(env: &dyn Environment, opts: GenerateOpts) -> DfxResult {
     env.get_cache().install()?;
 
     // Option can be None which means generate types for all canisters
-    let canister_names = config
+    let canisters_to_load = config
         .get_config()
         .get_canister_names_with_dependencies(opts.canister_name.as_deref())?;
+    let canisters_to_generate = canisters_to_load
+        .clone()
+        .into_iter()
+        .filter(|canister_name| {
+            !matches!(
+                config
+                    .get_config()
+                    .get_remote_canister_id(canister_name, &env.get_network_descriptor().name),
+                Ok(Some(_))
+            )
```

Comment:
There's a `is_remote_canister` method for convenience:
```suggestion
            config
                .get_config()
                .is_remote_canister(canister_name, &env.get_network_descriptor().name)
                .unwrap_or(false)
```